cluster/node-pools/worker-karpenter/provisioners.yaml

```diff
@@ -0,0 +1,125 @@
+{{- if eq .NodePool.Profile "worker-karpenter" }}
+---
+apiVersion: karpenter.sh/v1alpha5
+kind: Provisioner
```

Comment:
For easier rollout we could separate the actual node pool creation into a separate PR. (It would be similar to deploying an IngressController but not deploying any Ingress yet). Of course the final version must probably support deploying both at the same time for cluster creation.